simapp/v2/simdv2/cmd/commands.go

```diff
@@ -140,6 +147,24 @@ func InitRootCmd[T transaction.Tx](
 		return nil, err
 	}
 
+	grpcSrv := grpcServer.GrpcServer()
+
+	grpcgatewayServer, err := grpcgateway.New[T](
+		logger,
+		deps.GlobalConfig,
+		grpcSrv,
+		simApp.InterfaceRegistry(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, mod := range deps.ModuleManager.Modules() {
+		if gmod, ok := mod.(ModuleWithGRPCGateway); ok {
```

Comment:
let's use module.HasGRPCGateway (from types/modules), and can you add a `/TODO(@julienrbrt)` and link https://github.com/cosmos/cosmos-sdk/pull/22701#pullrequestreview-2470651390